internal/client/blueprints_test.go

```diff
@@ -1101,6 +1101,46 @@ func TestBlueprintFreezeV0(t *testing.T) {
 	require.NotEqual(t, "*", frozen.Blueprints[0].Blueprint.Modules[0].Version, "Wrong version in frozen blueprint module")
 }
 
+func TestBlueprintFreezeGlobsV0(t *testing.T) {
+	// Test needs real packages, skip it for unit testing
+	if testState.unitTest {
+		t.Skip()
+	}
+
+	bp := `{
+		"name": "test-freeze-blueprint-glob-v0",
+		"description": "TestBlueprintFreezeGlobsV0",
+		"version": "0.0.1",
+		"packages": [{"name": "tmux", "version": "*"},
+		{"name": "openssh-*", "version": "*"}]
+	}`
+
+	// Push a blueprint
+	resp, err := PostJSONBlueprintV0(testState.socket, bp)
+	require.NoError(t, err, "POST blueprint failed with a client error")
+	require.True(t, resp.Status, "POST blueprint failed: %#v", resp)
+
+	// Freeze the blueprint
+	frozen, api, err := FreezeBlueprintV0(testState.socket, "test-freeze-blueprint-glob-v0")
+	require.NoError(t, err, "Freeze blueprint failed with a client error")
+	require.Nil(t, api, "FreezeBlueprint failed: %#v", api)
+	require.Greater(t, len(frozen.Blueprints), 0, "No frozen blueprints returned")
+	require.Greater(t, len(frozen.Blueprints[0].Blueprint.Packages), 0, "No frozen packages returned")
+
+	var names []string
+	for _, p := range frozen.Blueprints[0].Blueprint.Packages {
+		assert.NotContains(t, p.Version, "*")
+		assert.NotContains(t, p.Version, "?")
```

Comment:
Since you are checking specifically for `?`, I think that it would make sense to use this pattern in the package name listed in the testing blueprint.